pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go

```diff
@@ -190,6 +191,31 @@ func (p *replicationFlowPlanner) containsInitialStreamAddresses() bool {
 	return len(p.initialStreamAddresses) > 0
 }
 
+func persistStreamIngestionPartitionSpecs(
+	ctx context.Context,
+	execCtx sql.JobExecContext,
+	ingestionJobID jobspb.JobID,
+	streamIngestionSpecs []*execinfrapb.StreamIngestionDataSpec,
+) error {
+	replicationPartitionInfoKey := "~replication-partition-specs.binpb"
+	err := execCtx.ExecCfg().InternalDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+		jobInfoStorage := jobs.InfoStorageForJob(txn, ingestionJobID)
+		specs := make([]*execinfrapb.StreamIngestionPartitionSpec, 0)
+		partitionSpecs := execinfrapb.StreamIngestionPartitionSpecs{Specs: specs}
+		for _, d := range streamIngestionSpecs {
+			for _, partitionSpec := range d.PartitionSpecs {
+				partitionSpecs.Specs = append(partitionSpecs.Specs, &partitionSpec)
+			}
+		}
+		specBytes, err := protoutil.Marshal(&partitionSpecs)
+		if err != nil {
+			return err
+		}
+		return jobInfoStorage.Write(ctx, replicationPartitionInfoKey, specBytes)
```

Comment:
this was a good callout. Unfortunately, the `Write` and `Read` methods in their current form were reserved for those files that we wanted to show up in the `Advanced Debugging` tab on the jobs details page. This is because they implicitly wrote an info row to a `~profiler/` prefixed info key. I have added a comment to this effect, but also pulled out the core of this method into its own utility so that other callsites that just want to write a file to job_info and read it back can use. Lets get https://github.com/cockroachdb/cockroach/pull/110035 merged before I rebase and fix this PR up?

Reply:
fixed